### PR TITLE
[Fixes #6739] Introduced wallet.settings hook

### LINKED ISSUE #6739: Implement a wallet extension hook
### User Story

As a developer, I want to add extra features in the wallet.

Proposed hook syntax:

```clojure
 hooks/wallet.settings.hello
 {:label     "Wallet action name"
  :view      [view]
  :on-click [event]}
```

### Description

*Type*: Feature

### PR DESCRIPTION
**Pull request by jeluard**
fixes #6739

### Summary:

Introduced an extension hook allowing to plug new screens in wallet/settings.

### Test

Make sure the following extension loads fine: https://status-im.github.io/pluto/try.html?hash=QmfPnUPvZ4af2H6Yj5CQZpFgkTevmeWb9DorwDtaFcSsCi

status: ready

### CODE REVIEW

File: src/status_im/extensions/registry.cljs [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
   (:require [clojure.string :as string]
-            [pluto.utils :as utils]
             [pluto.reader.hooks :as hooks]
             [re-frame.core :as re-frame]
             [status-im.accounts.update.core :as accounts.update]
             [status-im.i18n :as i18n]
             [status-im.utils.fx :as fx]
             [clojure.set :as set]))
 
+(defn- update-db [cofx {:keys [db]}]
```
Comment (yenda) [inline src/status_im/extensions/registry.cljs]: why duplicated update-db and merge-fx from fx.cljs here ?

**Review by yenda — DISMISSED** [review]
changes on abi-spec don't seem related and break the spec

**Comment by jeluard** [comment]
@yenda I made changes according to your comments

File: src/status_im/extensions/core.cljs [hunk 2]
```diff
@@ -148,15 +150,16 @@
...
 (defn button [{:keys [on-click]} label]
-  [button/secondary-button {:on-press #(re-frame/dispatch (on-click {}))} label])
+  [button/secondary-button (when on-click {:on-press #(re-frame/dispatch (on-click {}))}) label])
 
 (defn input [{:keys [on-change placeholder]}]
-  [react/text-input {:placeholder placeholder
-                     :style {:width "100%"}
-                     :on-change-text #(re-frame/dispatch (on-change {:value %}))}])
+  [react/text-input (merge {:placeholder placeholder
+                            :style {:width "100%"}}
```
Comment (flexsurfer) [inline src/status_im/extensions/core.cljs:159]: why `width "100%"` but not `flex 1` ?

File: src/status_im/ui/screens/wallet/settings/views.cljs [hunk 3]
```diff
@@ -38,3 +52,37 @@
...
                        :render-fn #(render-token % visible-tokens)}]]]))
+
+(defview settings-hook []
+  (letsubs [{:keys [title view]} [:get-screen-params :wallet-settings-hook]]
+    [react/view {:style {:flex 1 :background-color :white}}
+     [status-bar/status-bar {:type :modal-wallet}]
+     [toolbar/toolbar {:style wallet.styles/toolbar}
+      [toolbar/nav-text {:handler             #(do (re-frame/dispatch [:update-wallet])
+                                                   (re-frame/dispatch [:navigate-back]))
+                         :style               {:color :white}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/settings/views.cljs]: `colors/white`

File: src/status_im/ui/screens/wallet/settings/views.cljs [hunk 4]
```diff
@@ -38,3 +52,37 @@
       [list/flat-list {:data      (tokens/sorted-tokens-for (ethereum/network->chain-keyword network))
                        :key-fn    (comp str :symbol)
                        :render-fn #(render-token % visible-tokens)}]]]))
+
+(defview settings-hook []
+  (letsubs [{:keys [title view]} [:get-screen-params :wallet-settings-hook]]
+    [react/view {:style {:flex 1 :background-color :white}}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/settings/views.cljs]: `colors/white`

File: src/status_im/ui/screens/wallet/settings/views.cljs [hunk 5]
```diff
@@ -38,3 +52,37 @@
...
+     [status-bar/status-bar {:type :modal-wallet}]
+     [toolbar/toolbar {:style wallet.styles/toolbar}
+      [toolbar/nav-text {:handler             #(do (re-frame/dispatch [:update-wallet])
+                                                   (re-frame/dispatch [:navigate-back]))
+                         :style               {:color :white}
+                         :accessibility-label :done-button}
+       (i18n/label :t/done)]
+      [toolbar/content-title {:color :white}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/settings/views.cljs]: `colors/white`

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/extensions/core.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/extensions/core.cljs:159]: Don't quite remember.. But it's not new from this PR

**Comment by jeluard** [comment]
@flexsurfer Addressed your comments

**Comment by antdanchenko** [comment]
@jeluard please rebase the PR on latest develop, as result an :accessibility-label for recovery phrase will be in the PR's build and failed tests will pass

**Comment by jeluard** [comment]
@antdanchenko Thanks! Done

**Comment by antdanchenko** [comment]
thanks, in next several days we will rework tests so they will be picked from the same branch as a PR, instead of develop branch and the rebase step won't be necessary

**Comment by asemiankevich** [comment]
so as discussed with Julien i have checked very briefly that i can add new wallet option by installing this extension https://status-im.github.io/pluto/try.html?hash=QmWZZkVjtGdcoGTqYvN7SZ3FXbRmFhFqsQSCBsiZ3gaFXt

```
{meta {:name          "Hello"
       :description   "World"
       :documentation ""}

 views/view
 [text "Hello"]
 
 hooks/wallet.settings.hello
 {:label   "Test wallet settings"
  :title   "Option when hook is added"
  :view    [view]}}
```
image

Should be fine for sanity check. e2e fails due to new label , this is known and fixed already

<blockquote><div><strong><a href="https://status-im.github.io/pluto/index.html">Pluto · A grammar for data manipulation</a></strong></div><div>A grammar for data manipulation</div></blockquote>

**Review by goranjovic — APPROVED** [review]

**Pull request merged**